omnigibson/scenes/scene_base.py

```diff
@@ -668,18 +743,41 @@ def is_fluid_system(self, system_name):
         return isinstance(self.get_system(system_name, force_init=False), FluidSystem)
 
     def get_system(self, system_name, force_init=True):
+        """
+        Grab the system @system_name, and optionally initialize it if @force_init is set
+
+        Args:
+            system_name (str): Name of the system to grab
+            force_init (bool): Whether to force the system to be initialized and added to set of active_systems
+                if not already
+
+        Returns:
+            BaseSystem: Requested system
+        """
         # Make sure scene exists
         assert self.loaded, "Cannot get systems until scene is imported!"
-        # If system_name is not in REGISTERED_SYSTEMS, create from metadata
-        system = self.system_registry("name", system_name)
-        assert system is not None, f"System {system_name} not in system registry."
+        assert system_name in self._available_systems, f"System {system_name} is not a valid system name"
+        # If system is not initialized, initialize and add it to our registry
+        system = self._available_systems[system_name]
         if not system.initialized and force_init:
             system.initialize(scene=self)
+            self.system_registry.add(system)
         return system
 
+    def remove_system(self, system_name):
+        """
+        Clear the system @system_name and remove it from our set of active systems
+
+        Args:
+            system_name (str): Name of the system to remove
+        """
+        system = self.system_registry("name", system_name)
+        if system is not None:
+            system.clear()
```

Comment:
should we also remove it from the system registry?

agreed with Hang that we should also update the doc.

Reply:
think about unifying import_object/remove_object/get_system/remove_system

Reply:
done